template/index.html

```diff
@@ -26,16 +26,59 @@
     <div id="app">This is your fallback content in case JavaScript fails to load.</div>
     <!-- Todo: only include in production -->
     <script>
-      if('serviceWorker' in navigator) {
-        navigator.serviceWorker.register('/service-worker.js', { scope: '/' })
+      (function() {
+        'use strict';
+
+        // Check to make sure service workers are supported in the current browser,
+        // and that the current page is accessed from a secure origin. Using a
+        // service worker from an insecure origin will trigger JS console errors.
+        const isLocalhost = Boolean(window.location.hostname === 'localhost' ||
+            // [::1] is the IPv6 localhost address.
+            window.location.hostname === '[::1]' ||
+            // 127.0.0.1/8 is considered localhost for IPv4.
+            window.location.hostname.match(
+              /^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/
+            )
+          );
+
+        if ('serviceWorker' in navigator &&
+            (window.location.protocol === 'https:' || isLocalhost)) {
+          navigator.serviceWorker.register('service-worker.js')
```

Comment:
It's usually a good idea to wrap the `register()` in a `window.onload`, as per https://developers.google.com/web/fundamentals/instant-and-offline/service-worker/registration#conclusion

If there's a "better" event to listen for that's exposed as part of the Vue framework, like an event that's fired after all all the custom framework logic has completed, that would be another option to wrap it in.